scripts/scr_poststage.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+#
+# SCR allows you to spawn off checkpoint transfers "in the background"
+# that will finish some time after a job completes.  This saves you from
+# using your compute time to transfer checkpoints.  You can do this by
+# specifying XFER=BBAPI_POSTSTAGE for a storage descriptor in scr.conf.
+# Currently, this is only supported on on IBM burst buffer nodes.
+#
+# This script is to be run as a 2nd-half post-stage script on an IBM
+# system.  A 2nd-half post-stage script will run after all the job's burst
+# buffers transfers have finished (which could be hours later after the job
+# finishes).  This script will finalize any completed burst buffer checkpoint
+# transfers so that they're visible to SCR.
+#
+
+# Please fill in BINDIR AND PREFIX below:
+
+# The dir path containing the SCR binaries (scr_index and scr_flush_file)
+BINDIR=""
+
+# The dir path to your SCR PREFIX
+PREFIX="/p/gpfs1/`whoami`"
+
+# Path to where you want the scr_poststage.sh log
+LOGFILE=$PREFIX/scr_poststage.log
+
+[ -d $BINDIR ] && echo "bindir is dir" >> $LOGFILE
+
+if [ ! -d "$BINDIR" ] ; then
+	echo "Please specify BINDIR in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+if [ ! -d "$PREFIX" ] ; then
+	echo "Please specify PREFIX in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+export PATH="$PATH:$BINDIR"
+
+echo "$(date '+%m/%d/%y %H:%m:%S') Begin post_stage" >> $LOGFILE
+echo "Current index before finalizing:" >> $LOGFILE
+scr_index -l -p $PREFIX >> $LOGFILE
+
+# Finalize each dataset listed in the flush file
+for ID in $(scr_flush_file --dir $PREFIX --list-ckpt) ; do
+	CKPT="ckpt.$ID"
+	echo "Looking at dataset $ID ($CKPT)" >> $LOGFILE
+
+	if ! scr_flush_file --dir $PREFIX --need-flush $ID &>> $LOGFILE ; then
+		# Dataset is already flushed, skip it
+		continue
+	fi
+
+	echo "Finalizing transfer for dataset $ID ($CKPT)" >> $LOGFILE
+	if ! scr_flush_file -r -s $ID --dir $PREFIX &>> $LOGFILE ; then
+		echo "Error: Can't resume dataset $ID, rc=$?" >> $LOGFILE
+		continue
+	fi
+
+	echo "Writing summary for dataset $ID" >> $LOGFILE
+	if ! scr_flush_file -s $ID -S --dir $PREFIX &>> $LOGFILE ; then
+		echo "ERROR: Can't write summary for dataset $ID" >> $LOGFILE
+		continue
+	fi
+
+	echo "Adding checkpoint $CKPT to index" >> $LOGFILE
+	if ! scr_index -p $PREFIX --add=$CKPT &>> $LOGFILE ; then
+		echo "Couldn't add checkpoint $CKPT to index" >> $LOGFILE
+		continue
+	fi
+
+	echo "Setting current checkpoint to $CKPT" >> $LOGFILE
+	if ! scr_index -p $PREFIX --current=$CKPT &>> $LOGFILE ; then
+		echo "Couldn't set current checkpoint to $CKPT" >> $LOGFILE
+		continue
+	fi
```

Comment:
Not all datasets are checkpoints.  I don't remember what happens if we set a non-checkpoint to be ``--current``.  This is a reminder that I need to review that.